---
title: "Cancer Genomics"
author: "Adarsh Rajesh"
date: "11/26/19"
output: html_document
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```

## Cancer Genomics

Part 1:

Using: portal.gdc.cancer.gov

Q1. How many Cases (i.e. patient samples) have been found to have p53 mutations?

  - 4008

Q2. What are the top 6 misssense mutations found in this gene? 
HINT: Scroll down to the ‘TP53 - Protein’ section and mouse over the displayed plot. For example R175H is found in 156 cases.

  - Chr. 17 : C > T (R175H)
  - Chr. 17 : G > A (R273C)
  - Chr. 17 : C > T (R248Q)
  - Chr. 17 : C > T (R273H)
  - Chr. 17 : G > A (R248W)
  - Chr. 17 : G > A (R282W)

Q3. Which domain of the protein (as annotated by PFAM) do these mutations reside in?

  - The DNA binding domain

Q4. What are the top 6 primary sites (i.e. cancer locations such as Lung, Brain, etc.) with p53 mutations and how many primary sites have p53 mutations been found in? 
HINT: Clicking on the number links in the Cancer Distribution section will take you to a summary of available data accross cases, genes, and mutations for p53. Looking at the cases data will give you a ranked listing of primary sites.

  - The top six sites are: Breast, Brain, Ovary, Lung, Uterus, and Kidney
  - 25 Primary Sites
  
Q5. What is the most frequentely mutated position associated with cancer in the KRas protein (i.e. the amino acid with the most mutations)?

  - G12D

Q6. Are KRas mutations common in Pancreatic Adenocarcinoma (i.e. is the Pancreas a common ‘primary site’ for KRas mutations?).

  - Yes. It has been seen in 75.27% of cases.

Q6. What is the ‘TGCA project’ with the most KRas mutations?

  - TCGA-UCEC

Q7. What precent of cases for this ‘TGCA project’ have KRas mutations and what precent of cases have p53 mutations? 
HINT: Placing your mouse over the project bar in the Cancer Distribution panel will bring up a tooltip with useful summary data.

  - 25.09%
  - 

Q8. How many TGCA Pancreatic Adenocarcinoma cases (i.e. patients from the TCGA-PAAD project) have RNA-Seq data available?

  - 
  

# The GenomicDataCommons R package


```{r}
library(GenomicDataCommons)
library(TCGAbiolinks)
library(maftools)
```

And load the library

```{r}
status()
```

# Querying GDC from R

```{r}
projects <- getGDCprojects()
head(projects)
```

```{r}
cases_by_project <- cases() %>%
  facet("project.project_id") %>%
  aggregations()
head(cases_by_project)
```



```{r}
samp <- getSampleFilesSummary("TCGA-PAAD")
head(samp)
```
```{r}
query <- GDCquery(project="TCGA-PAAD",
                  data.category="Transcriptome Profiling",
                  data.type="Gene Expression Quantification")

ans <- getResults(query)
head(ans)
```

```{r}
View(ans)
```

```{r}
nrow(ans)
```
 
# Variant analysis with R

```{r}
maf.file <- GDCquery_Maf(tumor="PAAD", pipelines = "mutect")

```

```{r}
head(maf.file)

```

```{r}
vars = read.maf(maf = maf.file, verbose = FALSE)

```

```{r}
plotmafSummary(vars)

```

```{r}
oncoplot(maf = vars, top = 10)

```
```{r}
# Oncoplot for our top 10 most frequently mutated genes
pdf("oncoplot_panc.pdf")
oncoplot(maf = vars, top = 10, fontSize = 12)
dev.off()
```

```{r}
oncostrip(maf=vars, genes=c("KRAS", "TP53"))

```

```{r}
lollipopPlot(vars, gene='KRAS')
```

```{r}
lollipopPlot(vars, gene='TP53')

```


